File: react-native-navigation-master/lib/android/app/src/main/java/com/reactnativenavigation/options/OverlayOptions.java

package com.reactnativenavigation.options;

import com.reactnativenavigation.options.params.Bool;
import com.reactnativenavigation.options.params.NullBool;
import com.reactnativenavigation.options.parsers.BoolParser;

import org.json.JSONObject;

public class OverlayOptions {
    public Bool interceptTouchOutside = new NullBool();
    public OverlayAttachOptions overlayAttachOptions = new OverlayAttachOptions();
    public static OverlayOptions parse(JSONObject json) {
        OverlayOptions options = new OverlayOptions();
        if (json == null) return options;
        options.overlayAttachOptions = OverlayAttachOptions.parse(json.optJSONObject("attach"));
        options.interceptTouchOutside = BoolParser.parse(json,"interceptTouchOutside");
        return options;
    }
}

File: react-native-navigation-master/lib/android/app/src/main/java/com/reactnativenavigation/options/OverlayOptions.kt
package com.reactnativenavigation.options

import com.reactnativenavigation.options.params.NullText
import com.reactnativenavigation.options.params.Text
import com.reactnativenavigation.options.parsers.TextParser
import org.json.JSONObject
class OverlayAttachOptions{

    @JvmField
    var layoutId :Text = NullText()
    @JvmField
    var anchorId :Text = NullText()
    @JvmField
    var gravity:Text = NullText()

    fun hasValue() = layoutId.hasValue()
    override fun toString(): String {
        return "OverlayAttachOptions(layoutId=$layoutId, anchorId=$anchorId, gravity=$gravity)"
    }

    companion object{
        @JvmStatic
        fun parse(json: JSONObject?): OverlayAttachOptions {
            val overlayAttachOptions = OverlayAttachOptions()
            overlayAttachOptions.layoutId = TextParser.parse(json,"layoutId")
            overlayAttachOptions.anchorId = TextParser.parse(json?.optJSONObject("anchor"),"id")
            overlayAttachOptions.gravity = TextParser.parse(json?.optJSONObject("anchor"),"gravity")
            return overlayAttachOptions
        }
    }
}